### PR TITLE
chore: update internal lint dependencies

### PR DESCRIPTION
**Pull request by bmish**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

Update internal lint dependencies.

#### What changes did you make? (Give an overview)

Update internal lint dependencies. Note that several lint dependencies could not be updated fully because they have already dropped support for Node 14 which we still test against on CI. We can update these during the next major version bump. 

Changelogs:

* https://github.com/igorshubovych/markdownlint-cli/releases
* https://github.com/DavidAnson/markdownlint/commits/main
* https://github.com/gajus/eslint-plugin-jsdoc/releases
* https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/main/CHANGELOG.md#440-2022-07-02

Also enabling two new rules I implemented:

* [eslint-plugin/no-missing-message-ids](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/main/docs/rules/no-missing-message-ids.md)
* [eslint-plugin/no-unused-message-ids](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/main/docs/rules/no-unused-message-ids.md)

### CODE REVIEW

**Review by snitin315 — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -47,19 +47,19 @@ Prerequisites: [Node.js](https://nodejs.org/) (`^12.22.0`, `^14.17.0`, or `>=16.
 
 You can install ESLint using npm:
 
-```sh
+```shell
```
Comment (ddzz) [inline README.md:50]: Is there any benefit from using `shell` instead of `sh` here?

Comment (bmish) [reply README.md:50]: It's the same thing but I switched this to `shell` for consistency with all other 90 usages in the codebase.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**